kong/cmd/utils/nginx_signals.lua

```diff
@@ -84,13 +84,24 @@ function _M.start(kong_conf)
 
   log.debug("starting nginx: %s", cmd)
 
-  local ok, _, _, stderr = pl_utils.executeex(cmd)
-  if not ok then
-    return nil, stderr
+  if kong_conf.nginx_daemon == "on" then
+    -- running as daemon, so we capture command output to temp files using
+    -- the "executeex" method
+    local ok, _, _, stderr = pl_utils.executeex(cmd)
+    if not ok then
+      return nil, stderr
+    end
+    log.debug("nginx started")
+
+  else
+    -- not as a dameon, so we cannot redirect because in long running processes
+    -- the output will fill the disk. Use "execute" without redirection instead.
+    local ok, retcode = pl_utils.execute(cmd)
```

Comment:
Since we only target one Lua version (LuaJIT), there's no need to use `pl_utils.execute` here, one can just use `os.execute` directly (and handle the Lua 5.1 semantics of the results).

Reply:
Not a blocker by any means -- I just noted the added cognitive step (I had to look into Penlight to check if `utils.execute` did anything else surprising like `executeex` does; I wouldn't have to if it was `os.execute`)

Reply:
@hisham How about any side effects? I'm not very familiar with the underlying unix structures (was surprised by the bug at all), so that's what I'm mostly concerned about.

Reply:
@hishamhm 

> Not a blocker by any means -- I just noted the added cognitive step (I had to look into Penlight to check if `utils.execute` did anything else surprising like `executeex` does; I wouldn't have to if it was `os.execute`)

Had to go look for it, see [this PR](https://github.com/stevedonovan/Penlight/pull/244) by @bungle . We had issues before. We only support LuaJIT, but that behaves differently based on the 52 compat-flag (compile time).

So we learned the hard way (but apparently not hard enough since I didn't remember 😄 ).

Reply:
Agreed - better to keep `pl_utils.execute`, the compat flag was a nightmare to get out of...

We should be adding a test for the foreground mode, asserting that we see Nginx's stdout/stderr logs...